pandas/core/frame.py

```diff
@@ -4901,6 +4901,25 @@ def reorder_levels(self, order, axis=0):
     # ----------------------------------------------------------------------
     # Arithmetic / combination related
 
+    def _wrap_dispatched_op(self, result, other, func, axis=None):
+        """
+        Wrap the result of an arithmetic/comparison operation performed
+        via ops.dispatch_to_series in a properly-indexed DataFrame.
+
+        Parameters
```

Comment:
this is very similar to the _box_item_values / _box_col_values, if you can de-duplicate

Reply:
I see the resemblance, but it would need significant tinkering to make work, and neither is implemented in SparseDataFrame as it is.

Reply:
its worth a go here, this is adding wholly duplicated code

Reply:
```
        result = self._constructor(result, index=self.index, copy=False)
        result.columns = self.columns
        return result
```

```
    def _box_item_values(self, key, values):
        items = self.columns[self.columns.get_loc(key)]
        if values.ndim == 2:
            return self._constructor(values.T, columns=items, index=self.index)
        else:
            return self._box_col_values(values, items)

    def _box_col_values(self, values, items):
        """ provide boxed values for a column """
        klass = _get_sliced_frame_result_type(values, self)
        return klass(values, index=self.index, name=items, fastpath=True)
```

`_box_col_values` is going to return either a `Series` or `SparseSeries` (determined by call to `_get_sliced_frame_result_type`), so can't use that.

`_box_item_values` is expecting an np.ndarray for `values`, whereas we have a dict.  It is also expecting a key, which I guess we could pass all of `self.columns` for (?).  Even then, it is passing `columns` directly into the `DataFrame` constructor, which we are specifically _avoiding_ here.

I really don't see how this code is duplicated or what can be re-used.  Not saying you're wrong, just that you're going to have to explain it to me in appreciably more detail.

Reply:
ok i buy this.